documentation/aleo/03_language.md

```diff
@@ -429,10 +429,71 @@ finalize transfer_public_to_private:
 
 The following commands are supported in Aleo Instructions to provide additional program functionality.
 
-#### self.caller
-The `self.caller` command returns the address of the caller of the program.  
+#### self.signer
+
+The `self.signer` command returns the user address that orginated the transition.
 This can be useful for managing access control to a program.
-In the above example, the `transfer_public_to_private` function decrements the balance of the sender publicly using `self.caller`.
+In the above example, the `transfer_public_to_private` function decrements the balance of the sender publicly using `self.signer`.
+
+#### self.caller
+
+The `self.caller` command returns the address of the immediate caller of the program.
+
+#### Usage of `self.signer` and `self.caller`
+
+Consider the following program as a motivating example for the usage of `self.signer` and `self.caller`
+
+```
+import import.aleo;
+
+program parent.aleo;
+
+record who_called_parent:
+    owner as address.private;
+    gates as u64.private;
+    // self.signer, should be original invoker of the program calling this one
```

Comment:
```suggestion
    // The address of the caller.
```
It's not clear to me why this "should be the original invoker of the program".
`caller` is a private address within a record data type.
I can import this record into another program and set `caller` to any arbitrary address.
`caller` cannot be restricted to a specific address at this point in the program.